lib/src/dart/grammar.dart

```diff
@@ -742,7 +745,7 @@ class DartGrammarDefinition extends GrammarDefinition {
   Parser identifierPartLexicalToken() =>
       ref0(identifierStartLexicalToken) | ref0(digitLexicalToken);
 
-  Parser letterLexicalToken() => letter();
+  Parser letterLexicalToken() => letter() & (ref1(token, "?") | ref1(token, "!")).optional();
```

Comment:
I don't think this is correct: doesn't this make 'a?bcd` and `a!bcd` to be valid identifiers, but `abcd?`, `abcd!` or `abcd()!` still doesn't work?

Reply:
You are right, the fix was good enough for my immediate problem, but not correct. It was a bit tricky to place it right, but I think this should be better